src/components/organisms/SchedulePageModal/SchedulePageModal.tsx

```diff
@@ -0,0 +1,200 @@
+import React, { useEffect, useState, useMemo } from "react";
+import "./SchedulePageModal.scss";
+import { OnlineStatsData } from "types/OnlineStatsData";
+import firebase from "firebase";
+import { startOfDay, addDays, isWithinInterval, endOfDay } from "date-fns";
+import _ from "lodash";
+
+type OpenVenues = OnlineStatsData["openVenues"];
+type OpenVenue = OpenVenues[number];
+
+type VenueEvent = {
+  venue: OpenVenue["venue"];
+  event: OpenVenue["currentEvents"][number];
+};
+
+type DatedEvents = Array<{
+  dateDay: Date;
+  events: Array<VenueEvent>;
+}>;
+
+const DAYS_AHEAD = 7;
+
+interface PropsType {
+  show?: boolean;
+  onHide?: () => void;
+}
+
+export const SchedulePageModal: React.FunctionComponent<PropsType> = ({
+  show,
+  onHide,
+}) => {
+  const [openVenues, setOpenVenues] = useState<OpenVenues>();
+  const [, setLoaded] = useState(false);
+
+  useEffect(() => {
+    const getOnlineStats = firebase
+      .functions()
+      .httpsCallable("stats-getOnlineStats");
+    const updateStats = () => {
+      getOnlineStats()
+        .then((result) => {
+          const { openVenues } = result.data as OnlineStatsData;
+          setOpenVenues(openVenues);
+          setLoaded(true);
+        })
+        .catch(() => {}); // REVISIT: consider a bug report tool
+    };
+    updateStats();
+    const id = setInterval(() => {
+      updateStats();
+    }, 60 * 1000);
+    return () => clearInterval(id);
+  }, []);
+
+  const orderedEvents: DatedEvents = useMemo(() => {
+    if (!openVenues) return [];
+
+    const nowDay = startOfDay(new Date());
+
+    const allEvents = openVenues.reduce<Array<VenueEvent>>(
+      (acc, ov) => [
+        ...acc,
+        ...ov.currentEvents.map((event) => ({ venue: ov.venue, event })),
+      ],
+      []
+    );
+
+    const dates: DatedEvents = _.range(0, DAYS_AHEAD).map((idx) => {
+      const day = addDays(nowDay, idx);
+
+      return {
+        dateDay: day,
+        events: allEvents
+          .filter((ve) =>
+            // some events will span multiple days. Pick events for which `day` is between the event start and end
+            isWithinInterval(day, {
+              start: startOfDay(new Date(ve.event.start_utc_seconds * 1000)),
+              end: endOfDay(
+                new Date(
+                  (ve.event.start_utc_seconds +
+                    ve.event.duration_minutes * 60) *
+                    1000
+                )
+              ),
+            })
+          )
+          .sort(
+            (a, b) => a.event.start_utc_seconds - b.event.start_utc_seconds
+          ),
+      };
+    });
+
+    return dates;
+  }, [openVenues]);
+
+  const months = [
+    "Jan",
+    "Feb",
+    "Mar",
+    "Apr",
+    "May",
+    "Jun",
+    "Jul",
+    "Aug",
+    "Sep",
+    "Oct",
+    "Nov",
+    "Dec",
+  ];
+  const DateToString = (Date: Date) =>
```

Comment:
can we use something similar or based on https://github.com/sparkletown/sparkle/blob/staging/src/utils/time.js#L64 ?